File: codechef/contests/APRIL16/CHBLLNS.cpp
#include <cstdio>

using namespace std;

long long min(long long a, long long b, long long c){
	if (a <= b && a <= c) return a;
	if (b <= c) return b;
	return c;
}

long long max(long long a, long long b, long long c){
	if (a >= b && a >= c) return a;
	if (b >= c) return b;
	return c;
}

long long mid(long long a, long long b, long long c){
	return (a+b+c)-(min(a,b,c)+max(a,b,c));
}

long long solve(long long r, long long g, long long b, long long k){
	long long ra, ga, ba;
	ra = min(r,g,b);
	ga = mid(r,g,b);
	ba = max(r,g,b);
	if (ra >= k-1) return 3*(k-1)+1;
	if (ga >= k-1) return 3*ra + 2*(k-ra-1) + 1;
	return ra+ga+k;
}
int main(void){
	long long r, g, b, k, t;
	scanf("%lld",&t);
	while (t--){
		scanf("%lld %lld %lld",&r,&g,&b);
		scanf("%lld",&k);
		printf("%lld\n",solve(r,g,b,k));
	}

}

File: codechef/contests/APRIL16/CHBLLNS.py

# def equal(a,b):
# 	if a >= 0 and b >= 0:
# 		return True
# 	if a < 0 and b < 0:
# 		return True
# 	return False

# def secmin(r, g, b):
# 	if equal(r-g,g-b): return g
# 	if equal(g-r,r-b): return r
# 	return b

# def solve(r, g, b, k):
# 	fmin = min(r, g, b)
# 	smin = secmin(r, g, b)
# 	ans = 0
# 	if fmin < k-1 and smin < k-1:
# 		return fmin*3 + (smin-fmin)*2 + k-smin
# 	if fmin < k-1:
# 		return fmin*3 + 2*(k-fmin-1) + 1
# 	return 3*(k-1) + 1


# for _ in xrange(input()):
# 	a = map(int, raw_input().split())
# 	a.append(input())
# 	print solve(*a)


# def case():
# 	t = random.randint(1,9)
# 	print t
# 	for i in xrange(t):
# 		for j in xrange(3):
# 				print random.randint(1,9),
# 		print
# 		print random.randint(1,9)

def solve(r, g, b, k):
	a = [r, g, b]
	r = min(a)
	b = max(a)
	g = sum(a) - (min(a) + max(a))
	if r >= k-1:
		return 3*(k-1)+1
	if g >= k-1:
		return 3*r + (k-r-1)*2 + 1
	return r+g+k

for _ in xrange(input()):
	r, g, b = map(int, raw_input().split())
	k = input()
	print solve(r, g, b, k)